src/App.tsx

```diff
@@ -17,18 +59,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                filter={filter}
+                setQuery={(newQuery) => setQuery(newQuery)}
+                setFilter={(newFilter) => setFilter(newFilter)}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {hasLoader && <Loader />}
+              <TodoList
+                todos={preparedTodos}
+                setTodo={(todo) => setModalTodo(todo)}
+                modalTodo={modalTodo}
+              />
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {modalTodo && (
+        <TodoModal
+          todo={modalTodo}
+          clearModal={() => setModalTodo(undefined)}
```

Comment:
usually we try not pass `undefined`
```suggestion
          clearModal={() => setModalTodo(null)}
```